imod/mf6/utilities/dataset.py

```diff
@@ -46,3 +47,17 @@ def is_dataarray_none(datarray: Any) -> bool:
 def get_scalar_variables(ds: GridDataArray) -> list[str]:
     """Returns scalar variables in a dataset."""
     return [var for var, arr in ds.variables.items() if is_scalar(arr)]
+
+
+def convert_time_column(
+    ds: GridDataArray, simulation_start_time: np.datetime64, time_unit: str = "d"
+) -> GridDataArray:
+    if not "time" in ds.coords:
+        raise ValueError(
+            "cannot convert time column, because a time column could not be found"
+        )
+
+    ds["time"] = pd.Timestamp(simulation_start_time) + pd.to_timedelta(
+        ds["time"], unit=time_unit
+    )
+    return ds
```

Comment:
Nice, good that you are using ``pandas`` for this. The canonical way to assign a coordinate is with the [``assign_coords`` method](https://docs.xarray.dev/en/stable/generated/xarray.Dataset.assign_coords.html).

```suggestion
    time = pd.Timestamp(simulation_start_time) + pd.to_timedelta(
        ds["time"], unit=time_unit
    )
    return ds.assign_coords(time=time)
```

In this way you avoid accidentily adding ``"time"`` as a variable, instead of coordinate for example.